soliket/ccl.py

```diff
@@ -175,3 +200,61 @@ def get_CCL(self):
         :return: dict of results
         """
         return self._current_state['CCL']
+
+class Tester(Likelihood):
+    # Cross and auto data
+
+    auto_file: str = 'input/clgg_noiseless.txt'
+    cross_file: str = 'input/clkg_noiseless.txt'
+    #auto_file: str = 'input/clgg.txt'
+    #cross_file: str = 'input/clkg.txt'
+    dndz_file: str = 'input/dndz.txt'
+
+    params = {'b1': 1, 's1': 0.4} 
+
+    #ell_file: str = "/Users/Pablo/Code/SOLikeT/soliket/data/simulated_ccl/ell.npy"
+    #cl_file: str = "/Users/Pablo/Code/SOLikeT/soliket/data/simulated_ccl/cls.npy"
+    #dcl_file: str = "/Users/Pablo/Code/SOLikeT/soliket/data/simulated_ccl/dcls.npy"
+
+    def initialize(self):
+        self.ccl = __import__("pyccl")
+        #self.cl_data = np.load(self.cl_file)
+        #self.dcl_data = np.load(self.dcl_file)
+        #self.ell_data = np.load(self.ell_file)
+        data_auto = np.loadtxt(self.auto_file)
+        data_cross = np.loadtxt(self.cross_file)
+        self.dndz = np.loadtxt(self.dndz_file)
+
+        # Get data
+        self.ell_auto = data_auto[0]
+        self.cl_auto = data_auto[1]
+        self.cl_auto_err = data_auto[2]
+
+        self.ell_cross = data_cross[0]
+        self.cl_cross = data_cross[1]
+        self.cl_cross_err = data_cross[2]  
+
+    def get_requirements(self):
+        return {'CCL': {#"methods": {'theory': self._get_theory},
+                        "kmax": 10,
+                        "nonlinear": True}}
+
+    def logp(self, **pars):
+        cosmo = self.provider.get_CCL()['cosmo']
+        #cosmo = results['theory']
+
+        tracer_g = self.ccl.NumberCountsTracer(cosmo, has_rsd=False, dndz = self.dndz.T, 
+                                            bias =(self.dndz[:,0], pars['b1']*np.ones(len(self.dndz[:,0]))), 
+                                            mag_bias = (self.dndz[:,0], pars['s1']*np.ones(len(self.dndz[:,0])))
+                                            )
+        tracer_k = self.ccl.CMBLensingTracer(cosmo, z_source = 1060)
+
+        cl_gg = self.ccl.cls.angular_cl(cosmo, tracer_g, tracer_g, self.ell_auto) #+ 1e-7
+        cl_kg = self.ccl.cls.angular_cl(cosmo, tracer_k, tracer_g, self.ell_cross)        
+        
+        #cl_gg, cl_kg = results['theory']
+        delta = np.concatenate([cl_gg - self.cl_auto, cl_kg - self.cl_cross])
+        sigma = np.concatenate([self.cl_auto_err, self.cl_cross_err])
+        chi2 = delta**2/sigma**2.
+        #chi2 = np.dot(r, self.invcov.dot(r))
+        return -0.5 * np.sum(chi2)
```

Comment:
Since this is a gaussian likelihood, this likelihood should inherit from `GaussianLikelihood`, and define the appropriate `_get_data`, `_get_cov`, and `_get_theory` methods.

Reply:
(see some of the other likelihoods for examples, or ask me for more clarification)